.github/workflows/generated_files_sync.yaml

```diff
@@ -143,4 +143,28 @@ jobs:
           fi
           echo "Passed ${POM}"
         done
-        echo "All group IDs start with com.google.maps. Good."
\ No newline at end of file
+        echo "All group IDs start with com.google.maps. Good."
+
+  package_name_check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - name: Ensure no new invalid package name in Java files
+      shell: bash
+      run: |
+        # grep returns "1" (error) when no output
+        set +e
+        echo "Finding files matching '*main/java/google/*.java'"
+        # In past, we published Java classes with wrong package names (google.)
+        # due to improper java_package field in proto files (https://protobuf.dev/programming-guides/proto3/#options).
+        # This check excludes these existing files.
+        invalid_files=$(find . -wholename '*main/java/google/*.java' \
```

Comment:
Are packages starting with `google.` the only invalid packages? I think the script and the PR description should explicitly say that to avoid confusion.

Reply:
It's a typical error but it's possible to have other package name. I'll update the logic.